src/main/java/com/example/petbuddybackend/service/user/ClientService.java

```diff
@@ -47,6 +55,34 @@ public void createClientIfNotExist(JwtAuthenticationToken token) {
         }
     }
 
+    public ClientDTO getClient(String clientEmail) {
+        Client client = getClientByEmail(clientEmail);
+        userService.renewProfilePicture(client.getAccountData());
+        return clientMapper.mapToClientDTO(client);
+    }
+
+    public ClientComplexInfoDTO addFollowingCaretakers(String clientEmail, Set<String> caretakerEmails) {
+
+        Client client = getClientByEmail(clientEmail);
+        assertClientNotFollowingItself(client, caretakerEmails);
+        assertNewFollowingCaretakersAreNotAlreadyFollowed(client, caretakerEmails);
```

Comment:
Pytanie czy trzeba to sprawdzać. Myślę, że jak trafi się duplikat i chce mu znowu dać follow, to ostatecznie osoby które miały być zaobserwowane są zaobserwowane

Reply:
Niby tak, ale zwrocilbym informacje, ze opiekun jest juz zaobserwowany